### PR TITLE
[11.0][MIG] account_payment_term_extension

### PR DESCRIPTION
**Pull request by nachoserra-domatix**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please squash together all "OCA Transbot..." commits.

File: account_payment_term_extension/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# -*- coding: utf-8 -*-
```
Comment (melroy89) [inline account_payment_term_extension/__init__.py:1]: You can remove the utf-8 lines from the top in Python3. Just safe the file in utf8 format.

File: account_payment_term_extension/models/account_payment_term.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
```
Comment (melroy89) [inline account_payment_term_extension/models/account_payment_term.py:1]: You can remove the utf-8 lines from the top in Python3. Just safe the file in utf8 format.

File: account_payment_term_extension/tests/test_account_payment_term_multi_day.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
+# -*- coding: utf-8 -*-
```
Comment (melroy89) [inline account_payment_term_extension/tests/test_account_payment_term_multi_day.py:1]: same..

**Review by melroy89 — CHANGES_REQUESTED** [review]
You can remove the utf-8 lines from the top in Python3. Just safe the file in utf8 format.

**Review by pmartineznetquest — APPROVED** [review]
LGTM in these scenarios:
- a payment term with a single payment date. ok
- a payment term with 2 payment dates. ok
- changing from 1 payment term to another recalculates the Due Date. ok
- changin invoice partner changes and recalculates Due Date. ok

File: account_payment_term_extension/README.rst [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+#. Go to the menu Accounting > Configuration > Management > Payment Terms.
+
+To use multiple payment days, define for each payment term line which payment days apply, separated by spaces, commas or dashes.
+
+Usage
+=====
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/95/10.0
```
Comment (angelmoya) [inline account_payment_term_extension/README.rst:32]: 11.0

File: account_payment_term_extension/i18n/es.po [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: \n"
+"Language: es_ES\n"
+"Plural-Forms: nplurals=2; plural=(n != 1);\n"
+
+#. module: account_payment_term_extension
+#: model:account.payment.term,name:account_payment_term_extension.sixty_days_end_of_month
+#: model:account.payment.term,note:account_payment_term_extension.sixty_days_end_of_month
+msgid "60 Days End of Month"
+msgstr "60 dias al final del mes"
```
Comment (angelmoya) [inline account_payment_term_extension/i18n/es.po:26]: Días

File: account_payment_term_extension/i18n/es.po [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+#. module: account_payment_term_extension
+#: model:ir.ui.view,arch_db:account_payment_term_extension.view_payment_term_line_form
+msgid "Due Date Computation"
+msgstr "Cálculo de fecha de vencimiento"
+
+#. module: account_payment_term_extension
+#: model:ir.model.fields,field_description:account_payment_term_extension.field_account_payment_term_line_months
+msgid "Number of Months"
+msgstr "Numero de meses"
```
Comment (angelmoya) [inline account_payment_term_extension/i18n/es.po:46]: Número

File: account_payment_term_extension/i18n/es_ES.po [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: \n"
+"Language: es_ES\n"
+"Plural-Forms: nplurals=2; plural=(n != 1);\n"
+
+#. module: account_payment_term_extension
+#: model:account.payment.term,name:account_payment_term_extension.sixty_days_end_of_month
+#: model:account.payment.term,note:account_payment_term_extension.sixty_days_end_of_month
+msgid "60 Days End of Month"
+msgstr "60 dias al final del mes"
```
Comment (angelmoya) [inline account_payment_term_extension/i18n/es_ES.po:25]: Días

File: account_payment_term_extension/i18n/es_ES.po [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+
+#. module: account_payment_term_extension
+#: model:ir.ui.view,arch_db:account_payment_term_extension.view_payment_term_line_form
+msgid "Due Date Computation"
+msgstr "Cálculo de fecha de vencimiento"
+
+#. module: account_payment_term_extension
+#: model:ir.model.fields,field_description:account_payment_term_extension.field_account_payment_term_line_months
+msgid "Number of Months"
+msgstr "Numero de meses"
```
Comment (angelmoya) [inline account_payment_term_extension/i18n/es_ES.po:45]: Número

File: account_payment_term_extension/models/account_payment_term.py [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+        default=False,
+        help="Allows to apply a chronological order on lines.")
+
+    def apply_payment_days(self, line, date):
+        """Calculate the new date with days of payments"""
+        if line.payment_days:
+            payment_days = line._decode_payment_days(line.payment_days)
+            if payment_days:
+                new_date = None
+                payment_days.sort()
```
Comment (angelmoya) [inline account_payment_term_extension/models/account_payment_term.py:98]: decode_payment_days call days.sort() before return, so you can delete this line

**Review by angelmoya — APPROVED** [review]
Some remarks but LGTM

**Comment by pedrobaeza** [comment]
@NachoSerra are you going to finish this?

**Comment by pedrobaeza** [comment]
Superseeded by #375

**Pull request closed without merging**